connectors/byoc.py

```diff
@@ -107,31 +106,34 @@ async def save(self, connector):
             doc=document,
         )
 
+    async def heartbeat(self, doc_id):
+        await self.update(doc_id=doc_id, doc={"last_seen": iso_utc()})
+
     async def update_filtering_validation(
         self, connector, validation_result, validation_target=ValidationTarget.ACTIVE
     ):
-        doc_to_update = deepcopy(connector.doc_source)
+        filtering = connector.filtering.to_list()
 
-        for filter_ in doc_to_update.get("filtering", []):
+        for filter_ in filtering:
             if filter_.get("domain", "") == Filtering.DEFAULT_DOMAIN:
                 filter_.get(e2str(validation_target), {"validation": {}})[
                     "validation"
                 ] = validation_result.to_dict()
 
         await self.client.update(
-            index=CONNECTORS_INDEX,
+            index=self.index_name,
             id=connector.id,
-            doc=doc_to_update,
+            doc={"filtering": filtering},
             retry_on_conflict=RETRY_ON_CONFLICT,
         )
 
-    def build_docs_query(self, native_service_types=None, connectors_ids=None):
+    async def supported_connectors(self, native_service_types=None, connector_ids=None):
```

Comment:
cc @vidok - I know you added the `build_docs_query` originally and passed the query inside a different method. What was the goal of original design and how do you feel about this change?

I'm not against it - it looks okay, but want to understand a bit more about it

Reply:
Initially, I wanted to rename `build_docs_query` to `supported_connectors_query`, because `build_docs_query` is like a general purpose query building method, but it's not.

Then I realized even `supported_connectors_query` was only used by the method `supported_connectors`.

So In `supported_connectors`, it simply does 2 things, build the query, and then call `get_all_docs`.